specification/network/resource-manager/Microsoft.Network/stable/2020-06-01/customIpPrefix.json

```diff
@@ -0,0 +1,404 @@
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/customIpPrefixes": {
+      "get": {
+        "tags": [
+          "CustomIPPrefixes"
+        ],
+        "operationId": "CustomIPPrefixes_List",
+        "description": "Gets all custom IP prefixes in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of CustomIPPrefix resources.",
+            "schema": {
+              "$ref": "#/definitions/CustomIPPrefixListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List resource group Custom IP prefixes": {
+            "$ref": "./examples/CustomIpPrefixList.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CustomIPPrefixPropertiesFormat": {
+      "properties": {
+        "cidr": {
+          "type": "string",
+          "description": "The prefix range in CIDR notation. Should include the start address and the prefix length. Should be a subset of its master custom IP Prefix parent CIDR."
+        },
+        "masterCustomIPPrefix": {
+          "type": "string",
+          "description": "The name of the master custom IP Prefix parent."
+        },
+        "commissionedState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The commissioned state of the Custom IP Prefix. Possible values are: 'Provisioning', 'Provisioned', 'Commissioning', 'Commissioned', 'Decommissioning', 'Deprovisioning'.",
+          "enum": [
+            "Provisioning",
+            "Provisioned",
+            "Commissioning",
+            "Commissioned",
+            "Decommissioning",
+            "Deprovisioning"
+          ],
+          "x-ms-enum": {
+            "name": "CommissionedState",
+            "modelAsString": true
+          }
+        },
+        "publicIpPrefixes": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ReferencedPublicIPPrefix"
```

Comment:
Please use `./network.json#/definitions/SubResource` which is an ID reference to avoid duplicating definitions and remove `ReferencedPublicIPPrefix` from the file

Reply:
fixed